include/nbl/builtin/hlsl/emulated/float64_t_impl.hlsl

```diff
@@ -182,26 +175,26 @@ NBL_CONSTEXPR_INLINE_FUNC bool operatorLessAndGreaterCommonImplementation(uint64
             if (lhsSign == rhsSign)
                 return false;
 
-            if (OpType == OperatorType::LESS)
-                return lhs < rhs;
-            else
-                return lhs > rhs;
+            Op compare;
+            return compare(lhs, rhs);
         }
         if (emulated_float64_t_impl::areBothZero(lhs, rhs))
             return false;
     }
 
-    const uint64_t lhsSign = ieee754::extractSign(lhs);
-    const uint64_t rhsSign = ieee754::extractSign(rhs);
+    const uint64_t lhsSign = ieee754::extractSignPreserveBitPattern(lhs);
+    const uint64_t rhsSign = ieee754::extractSignPreserveBitPattern(rhs);
 
     // flip bits of negative numbers and flip signs of all numbers
-    uint64_t lhsFlipped = lhs ^ ((0x7FFFFFFFFFFFFFFFull * lhsSign) | ieee754::traits<float64_t>::signMask);
-    uint64_t rhsFlipped = rhs ^ ((0x7FFFFFFFFFFFFFFFull * rhsSign) | ieee754::traits<float64_t>::signMask);
-
-    if (OpType == OperatorType::LESS)
-        return lhsFlipped < rhsFlipped;
-    else
-        return lhsFlipped > rhsFlipped;
+    if (lhsSign)
+        lhs ^= 0x7FFFFFFFFFFFFFFFull;
+    if (rhsSign)
+        rhs ^= 0x7FFFFFFFFFFFFFFFull;
+    lhs ^= ieee754::traits<float64_t>::signMask;
+    rhs ^= ieee754::traits<float64_t>::signMask;
```

Comment:
this was a bitwise OR not XOR in the old code?

Reply:
xor, i want to flip sign bits

Reply:
aah ok I see